Framework/TestHelpers/src/ComponentCreationHelper.cpp

```diff
@@ -696,6 +696,50 @@ createMinimalInstrument(const Mantid::Kernel::V3D &sourcePos,
   return instrument;
 }
 
+/*
+An instrument creation helper allowing you to include/omit
+source/detector/sample. From createMinimalInstrument.
+*/
+Instrument_sptr createInstrumentWithOptionalComponents(
```

Comment:
👍 Good thinking!

Reply:
I would implement this method slightly differently:

* Internally call `createMinimalInstrument()` this way you know you get a `valid` instrument to begin with. Note that if you do not need the positions for source, sample, detector for the returned instrument, just hard-code some input values for those here.
* Implement use Component::remove() to remove the components you do now want in the instrument. See pseudo code below.

```cpp
auto validInstrument = createMinimalInstrument(V3D{0,0,-10}, V3D{0,0,0}, V3D{0,0,1});
auto sourcePtr = validInstrument.source();
auto samplePtr = validInstrument.sample();
if(!hasSample){
   validInstrument.remove(samplePtr.get());
}
if(!hasSource){
   validInstrument.remove(sourcePtr.get());
}
```